docs/admin/serverless/servicemesh/README.md

```diff
@@ -263,4 +295,123 @@ kubectl exec -it sleep-6d6b49d8b8-6ths6 -- curl -v sklearn-iris-burst-predictor-
+
+Specially under a mesh-wide strict mTLS setup, you will notice that workloads that are not part of the mesh will become blocked from calling InferenceServices. The easiest way to fix this is by [disabling the KServe top level virtual service](#disable-top-level-vs). However, if you need the KServe top level virtual service, you can still have a working hybrid environment by configuring KServe with its dedicated local Istio Gateway.
+
+First, you will need to patch Knative gateway, as shown [above for turning on mesh-wide strict mTLS](#mesh-wide-mtls). Then, you will need to create the following resources:
+
+```yaml
+apiVersion: networking.istio.io/v1beta1
+kind: Gateway
+metadata:
+  name: kserve-local-gateway
+  namespace: istio-system
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+    - hosts:
+        - '*'
+      port:
+        name: http
+        number: 8082
+        protocol: HTTP
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: kserve-local-gateway
+  namespace: istio-system
+spec:
+  ports:
+    - name: http
+      protocol: TCP
+      port: 80
+      targetPort: 8081
+  selector:
+    istio: ingressgateway
+```
+
+Finally, you need to edit **ingress** config in the inferenceservice ConfigMap to have the following:
+
+```bash
+kubectl edit configmap/inferenceservice-config --namespace kserve
+
+ingress : |- {
+    "knativeLocalGatewayService": "knative-local-gateway.istio-system.svc.cluster.local",
+    "localGateway":               "istio-system/kserve-local-gateway",
+    "localGatewayService":        "kserve-local-gateway.istio-system.svc.cluster.local"
+}
+```
+
+After restarting KServe controller, the needed Services and VirtualServices will be reconfigured and workloads outside the mesh would be unblocked from calling InferenceServices.
+
+## Securing the network border with TLS
+If you require TLS over all the network, the easiest is to enable mesh-wide strict mTLS in Istio. Though, communications are secure only for workloads inside the service mesh and the mesh-border is the weak point: if the Istio ingress gateway is not setup with TLS, data arriving at or departing from the mesh would still be plain-text.
+
+If you need the mesh-border to also be secured with TLS:
+
+* For requests coming from outside the cluster (i.e. public endpoints), you can follow [Knative external domain encryption guide](https://knative.dev/docs/serving/encryption/external-domain-tls/)
+* For requests coming from workloads inside the cluster which are not part of the service mesh, you may use [Knative cluster local encryption](https://knative.dev/docs/serving/encryption/encryption-overview/#cluster-local-encryption), although it is still in experimental state. It would, also, require you to [disable the KServe top level virtual service](#disable-top-level-vs) to ensure there is no plain-text port-of-entry.
```

Comment:
```suggestion
* For requests coming from workloads inside the cluster that are not part of the service mesh, you may use [Knative cluster local encryption](https://knative.dev/docs/serving/encryption/encryption-overview/#cluster-local-encryption), although it is still in an experimental state. It would, also, require you to [disable the KServe top-level virtual service](#disable-top-level-vs) to ensure there is no plain-text port-of-entry.
```